From 8d5422151aa78e8116ab562961ffeb598560b694 Mon Sep 17 00:00:00 2001
From: Kedareswara rao Appana <appana.durga.rao@xilinx.com>
Date: Wed, 23 Sep 2015 13:21:29 +0530
Subject: [PATCH 0585/1566] net: ethernet: xilinx: Update link state change
 properly in the driver

The last link state in the driver in adjust_link() was not being updated unless
There was the speed was set which could then cause the mac speed error message
Forever at a periodic rate.This patch fixes this issue.

Reported-by: John Linn <john.linn@xilinx.com>
Signed-off-by: Kedareswara rao Appana <appanad@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/net/ethernet/xilinx/xilinx_axienet_main.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
index c099532..1040ef8 100644
--- a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
+++ b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
@@ -580,12 +580,13 @@ static void axienet_adjust_link(struct net_device *ndev)
 			}
 
 			axienet_iow(lp, XAE_EMMC_OFFSET, emmc_reg);
-			lp->last_link = link_state;
 			phy_print_status(phy);
 		} else {
 			netdev_err(ndev,
 				   "Error setting Axi Ethernet mac speed\n");
 		}
+
+		lp->last_link = link_state;
 	}
 }
 
-- 
1.7.5.4

